src/bosh-director/lib/bosh/director/api/instance_manager.rb

```diff
@@ -30,7 +30,8 @@ def find_instances_by_deployment(deployment)
       # @return [Array] List of instances that matched the filter
       # @raise [InstanceNotFound]
       def filter_by(deployment, filter)
-        InstanceLookup.new.by_filter(filter.merge(deployment_id: deployment.id))
+        return InstanceLookup.new.by_filter(filter.merge(deployment_id: deployment.id)) if filter.is_a?(Hash)
```

Comment:
I'm not sure of Sequel syntax, but instead of this dual-call thing which looks a little confusing and fragile, do you know if it's possible to convert filter hashes to arrays with something like the following? Mainly trying to think how to only have a single `InstanceLookup.new.by_filter` and `deployment_id` filter.

    filter = [filter] if filter.is_a?(Hash)